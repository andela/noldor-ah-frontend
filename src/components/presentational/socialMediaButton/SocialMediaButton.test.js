import React from 'react';
import { shallow } from 'enzyme';
import SocialButton from './SocialMediaButton';

describe('SocialButton Shallow rendering', () => {
  const handleSocialButton = jest.fn();
  const wrapper = shallow(
    <SocialButton
      className="social-btn"
      fonticon="fa-google-plus-g"
      onClick={handleSocialButton}
      href="somewebsite.com"
    />
  );

  it('should render button with the appropriate props', () => {
    expect(wrapper.find('social-btn').exists());
  });

  it('matches the snapshot', () => {
    const tree = shallow(
      <SocialButton
        className="social-btn"
        fonticon="fa-google-plus-g"
        onClick={handleSocialButton}
        href="somewebsite.com"
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
