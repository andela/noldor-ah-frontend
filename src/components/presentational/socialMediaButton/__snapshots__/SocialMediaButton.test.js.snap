// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButton Shallow rendering matches the snapshot 1`] = `
<div
  onClick={[MockFunction]}
>
  <a
    href="somewebsite.com"
  >
    <button
      className="social-btn"
    />
  </a>
</div>
`;
